src/protocols/echo/Echo.h

```diff
@@ -36,11 +36,15 @@
 
 namespace chip {
 namespace Protocols {
+namespace Echo {
 
-enum
+/**
+ * Echo Protocol Message Types
+ */
+enum class MsgType
```

Comment:
```suggestion
enum class MsgType: uint8_t
```
which will avoid the need for the static casts above?

Reply:
No, it won't help.

Reply:
I think it is always good to specify the underlying type for scoped enum if the underlying type is fixed, but explicit conversion is still needed.

Reply:
let's fix the API, then?

Reply:
@rwalker-apple 
How to fix the API? we still need to explicitly convert the enum class MsgType : unit8_t to unit8_t, otherwise, we will hit the following error:

TransportLayerTests.TestSecureSessionMgr.cpp.o
../../src/transport/tests/TestSecureSessionMgr.cpp: In function ‘void {anonymous}::SendEncryptedPacketTest(nlTestSuite*, void*)’:
../../src/transport/tests/TestSecureSessionMgr.cpp:254:66: error: cannot convert ‘chip::Protocols::Echo::MsgType’ to ‘uint8_t’ {aka ‘unsigned char’}
  254 |     payloadHeader.SetMessageType(chip::Protocols::Echo::MsgType::EchoRequest);
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~

Reply:
Ideally I think we should tie the tuples of protocol + messagetype, so that we can ensure that echo types always go with echo protocols (even if other protocols have message types with the same ids).

Unsure how to do that quickly, so for now I would propose:

```cpp
....
namespace MsgType {
  constexpr uint8_t Request = 0;
} // namespace MsgType
```

so then I can use `chip::Protocols::Echo::MsgType::Request` (I also renamed echorequest to request because Echo::...::EchoRequest seems to stutter a bit).

Reply:
I like this idea 
@rwalker-apple do you agree?

Reply:
anything that avoids the cast of the parameter to SendMessage()